src/build/build.ts

```diff
@@ -120,102 +82,95 @@ export function build(options?: BuildOptions, config?: ProjectConfig): Promise<a
       Object.assign(optimizeOptions.js, options.js);
     }
 
-    let gulpfile = getGulpfile();
-    let userTransformers = gulpfile && gulpfile.transformers;
-    if (userTransformers) {
-      logger.debug(`${userTransformers.length} transformers found in gulpfile`);
-    }
-
-    let sourcesProject = new HtmlProject();
-    let depsProject = new HtmlProject();
-    let analyzer = new StreamAnalyzer(root, entrypoint, shell, fragments);
-    let bundler = new Bundler(root, entrypoint, shell, fragments, analyzer);
-
     logger.info(`Building application...`);
+
     logger.debug(`Reading source files...`);
-    let sourcesStream =
-      vfs.src(allSources, {cwdbase: true, allowEmpty: true})
-        .pipe(sourcesProject.split)
-        .pipe(compose(userTransformers))
-        .pipe(optimize(optimizeOptions))
-        .pipe(sourcesProject.rejoin);
+
+
+    let sourcesStream = polymerProject.sources()
+      .pipe(polymerProject.splitHtml())
+      .pipe(optimize(optimizeOptions))
+      .pipe(polymerProject.rejoinHtml());
 
     logger.debug(`Reading dependencies...`);
-    let depsStream =
-      vfs.src(dependencies, {cwdbase: true, allowEmpty: true})
-        .pipe(depsProject.split)
-        .pipe(optimize(optimizeOptions))
-        .pipe(depsProject.rejoin);
+    let depsStream = polymerProject.dependencies()
+      .pipe(polymerProject.splitHtml())
+      .pipe(polymerProject.rejoinHtml());
 
-    let allFiles = mergeStream(sourcesStream, depsStream)
+    let buildStream = mergeStream(sourcesStream, depsStream)
       .once('data', () => { logger.debug('Analyzing build dependencies...'); })
-      .pipe(analyzer);
+      .pipe(polymerProject.analyze);
 
     let serviceWorkerName = 'service-worker.js';
 
-    let unbundledPhase = new ForkedVinylStream(allFiles)
+    let unbundledPhase = forkStream(buildStream)
       .once('data', () => { logger.info('Generating build/unbundled...'); })
       .pipe(
         gulpif(
           options.insertDependencyLinks,
-          new PrefetchTransform(root, entrypoint,
-                shell, fragments, analyzer)
+          new PrefetchTransform(polymerProject.root, polymerProject.entrypoint,
```

Comment:
I intentionally haven't touched this in this PR, but totally agree on this point.